lstchain/io/io.py

```diff
@@ -670,3 +671,38 @@ def recursive_copy_node(src_file, dir_file, path):
             recursive_path = os.path.join(recursive_path, p)
 
 
+def write_calibration_data(writer,mon_index, mon_event, new_ped=False, new_ff=False):
+    mon_event.pedestal.prefix = ''
+    mon_event.flatfield.prefix = ''
+    mon_event.calibration.prefix = ''
+    mon_index.prefix = ''
+
+
+    # update index
+    if new_ped:
+        mon_index.pedestal_id += 1
```

Comment:
I find it a bit surprising that a `write_` function mutates its input. This should not happen!

Reply:
Also, This function depends on two booleans that make ifs inside this. This is a clear sign that this function should just be split in two, one to write a new pedestal and one to write a new flat fielding.

Reply:
yes, all this is ugly, but I expect this module will be soon replaced by a clean tool. I just try to reduce the number of function calls inside  it (it is already long enough)